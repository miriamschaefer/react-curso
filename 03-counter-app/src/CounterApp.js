import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 0}) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);
    const handleSubstract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);


    return (
        <>
            <h1>CounterApp</h1>
            <p>{counter}</p>
            <button onClick={handleAdd}>+</button>
            <button onClick={handleSubstract}>-</button>
            <button onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;