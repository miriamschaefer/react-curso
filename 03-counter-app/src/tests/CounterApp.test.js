import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';
import { stringify } from 'json5';

describe('tests in CounterApp component', () => {

    let wrapper; // in this moment, its value is undefined, you could initialize it as the content so you don't lose vscode autocomplete even though it's not as clean.

    beforeEach(() => {
        wrapper = shallow(<CounterApp/>);
    });

    test('should render <CounterApp /> component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should show a value of 100', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>);
        const element = wrapper.find('p').text().trim();
        expect(element).toBe(`${value}`);
    });

    test('should increment value by 1', () => {
        wrapper.find('button').at(0).simulate('click');
        const element = wrapper.find('p').text().trim();
        expect(element).toBe('1');
    })

    test('should decrement value by 1', () => {
        wrapper.find('button').at(1).simulate('click');
        const element = wrapper.find('p').text().trim();
        expect(element).toBe('-1');
    });

    test('should reset value to prop value', () => {
        const value = 105;
        const wrapper = shallow(<CounterApp value={value}/>);
        wrapper.find('button').at(2).simulate('click');
        const element = wrapper.find('p').text().trim();
        expect(element).toBe(stringify(value));


    })

})