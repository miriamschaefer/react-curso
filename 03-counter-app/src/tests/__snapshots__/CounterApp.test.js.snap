// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in CounterApp component should render <CounterApp /> component 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
